llvm/lib/Target/DirectX/DXILLegalizePass.cpp

```diff
@@ -605,6 +649,7 @@ class DXILLegalizationPipeline {
     LegalizationPipeline[Stage1].push_back(legalizeMemCpy);
     LegalizationPipeline[Stage1].push_back(removeMemSet);
     LegalizationPipeline[Stage1].push_back(updateFnegToFsub);
+    LegalizationPipeline[Stage1].push_back(legalizeLoadStoreOnArrayAllocas);
```

Comment:
Are you sure its ok to put this in stage 1? We already do IR replacement for loads and stores in this stage just want to be cautions about adding the same instruction twice to `ToRemove`.

Reply:
To be clear I think if you have a load\store with an i8. this could crash because `fixI8UseChain` will also add the load\store to `ToRemove`.

Reply:
Good catch. It does error when there is an i8 load/store directly on an alloca.
I should move this to stage 2 then?

Reply:
That would be the easiest. I have thought about changing this to a set, but while i think that would fix the crashing I dunno if that would result in correct transformation because we would be making assumptions on stale results.